### DISCUSSION TITLE
One-Time Password - Enter one-time password (Lost your Device?)

### DISCUSSION THREAD

**Discussion by MondoCat** [authelia/authelia#6160] [Q&A] [answered]
So, when going to log in, it states: "One-Time Password - Enter one-time password (Lost your Device?)"

Which implies that a one time password was sent to your device.

After waiting a long period of time and checking Nextcloud app, i got nothing.

Turns out it is an EMAIL one-time request to set UP a passcode, NOT a device password.

Please update the wording to say "One-Time Email authentication passcode" the first time you set it up. Not "Device passcode", as there is no passcode going to your device at that point.

**Comment by james-d-elliott** [comment]
The notification that appears in the browser is exactly `An email has been sent to your address to complete the process` (see [here](https://github.com/search?q=repo%3Aauthelia%2Fauthelia+%22An+email+has+been+sent+to+your+address+to+complete+the+process%22&type=code)). There is no mention of `device passcode` anywhere in the source (see [here](https://github.com/search?q=repo%3Aauthelia%2Fauthelia+%22device+passcode%22&type=code)).

Could you provide a screenshot of exactly what is confusing to you. Naming it a `one-time email authentication passcode` will just not happen as this is not what it is (would also be interested in a screenshot of a code that fits that input that has been emailed to you because we quite literally do not do this). As I've already partly explained the One-Time Password must be registered and stored on a device with a compatible app. The only part of this that uses email is the registration, not the technology itself.

**Reply by MondoCat** [reply]
It does not send a passcode when there is nothing to send a passcode to.

It sends you an email to register on authy.

Because authy is not set up.

So when you have not set up authy yet, it should NOT prompt for a passcode yet, because there is no passcode.


It should prompt "check your email to set up authy and then input passcode"


Make a new account yourself. You won't get a passcode if you have no passcode system set up. You will get an email to set one up... but the website does not tell you to check email to set one up.

https://cdn.discordapp.com/attachments/1008178258199711776/1164278968111874058/image.png?ex=6542a249&is=65302d49&hm=da4529e4885c433cffa69a06703b76beb062bbfe56feb9d17c2430610cdf79b9&


See, nowhere does it say it's sending an email to set up authy to get the passcode.

It just prompts for the passcode.

Which doesn't exist yet.


After authy has been set up, yes this is the correct wording. But not before.

**Reply by james-d-elliott** [reply, marked as answer]
Of course. If you have not set one up it's not going to prompt for a code. You have to set it up. When you click the link to do so it absolutely does notify you that you need to check your email. See the below screenshots.

![image](https://github.com/authelia/authelia/assets/3903683/4789425c-21e6-4cb7-a9e0-434515a3b5b0)

![image](https://github.com/authelia/authelia/assets/3903683/b46849a6-800b-44d6-9b57-c0ce2ab03292)

Setup:

![image](https://github.com/authelia/authelia/assets/3903683/76327358-a195-49fa-8d32-e20f5cbd70b7)

After setup:

![image](https://github.com/authelia/authelia/assets/3903683/3866e39f-9ae2-4b46-917b-8d2b32a95876)

![image](https://github.com/authelia/authelia/assets/3903683/94edda60-03d6-4efa-af50-8533b8154a57)
